### ISSUE TITLE
Some classpath inclusions cause errors when testing exceptions

### ISSUE THREAD

**Issue by sheluchin** [clojure-emacs/cider-nrepl#760]
## Expected behavior
Tests should be able to run regardless of what is on the classpath. If code that is on the classpath has not been evaluated, it should have no impact on test execution.

## Actual behavior
Placing broken code in `:paths` or `:extra-paths` causes test with exceptions to hang/produce an error.

## Steps to reproduce the problem
Clone https://github.com/oakes/lightmod and put the directory in your `:extra-paths`.

deps.edn
```edn
{:paths ["src" "oakes"]
 :deps {org.clojure/clojure {:mvn/version "1.11.1"}}}
```

src/foo.clj
```clojure
(ns foo
  (:require [clojure.test :as t]))

(t/deftest foo-test
  (t/is (= 1 (throw (Exception.)))))
```

Execute the test to produce the error. See screenshot:
<meta http-equiv="content-type" content="text/html; charset=utf-8"><img alt="2022-09-21 5 47 18" src="https://user-images.githubusercontent.com/6941/191362462-58050d70-04f1-4b92-89de-c9706d2e3978.png" style="max-width: 100%;" width="1001">

I'm not exactly sure why putting this particular repo causes such an error. I started with a list of about 100 different Clojure projects and it wasn't until I included this one that I ran into this problem. I am presuming there is some broken clj code in this repo, but I don't know the exact cause.

## Environment & Version information

I originally reported this as an [issue](https://github.com/liquidz/vim-iced/issues/445#issuecomment-1252911923) in vim-iced (which uses cider-nrepl to run tests), but with @liquidz help it seems like we were able to isolate it to cider-nrepl. 

### cider-nrepl version

cider/cider-nrepl {:mvn/version "0.28.5"}

### Java version

Clojure 1.11.1
OpenJDK 64-Bit Server VM 11.0.16+8-post-Ubuntu-0ubuntu122.04

### Operating system

Ubuntu 22.04

**Comment by vemv** [comment]
Hi! Thanks for the report.

> Tests should be able to run regardless of what is on the classpath.

This assumption isn't correct. You should have correct code in your classpath. Code that cannot be loaded by the Clojure compiler itself (which is presumably what is happening) cannot be worked with by cider-nrepl.

Please start by using cider-nrepl latest, which bundles one related bugfix: https://github.com/clojure-emacs/orchard/releases/tag/v0.10.0

If the issue persists, please open a more specific issue.

Cheers - V

**Comment by sheluchin** [comment]
Thanks, @vemv.

Just wanted to confirm that the issue is fixed after vim-iced bumped cider-nrepl to 0.28.6.
